pylearn2/datasets/hdf5.py

```diff
@@ -2,338 +2,485 @@
-            settings = list(propfaid.get_cache())
-            settings[2] = cache_size
-            propfaid.set_cache(*settings)
-            fid = h5py.h5f.open(filename, fapl=propfaid)
-            self._file = h5py.File(fid)
-        else:
-            self._file = h5py.File(filename)
-        if X is not None:
-            X = self.get_dataset(X, load_all)
-        if topo_view is not None:
-            topo_view = self.get_dataset(topo_view, load_all)
-        if y is not None:
-            y = self.get_dataset(y, load_all)
-
-        super(HDF5Dataset, self).__init__(X=X, topo_view=topo_view, y=y,
-                                          **kwargs)
-
-    def _check_labels(self):
+    def __new__(cls, filename, X=None, topo_view=None, y=None, load_all=False,
+                cache_size=None, sources=None, spaces=None, aliases=None,
+                use_h5py='auto', **kwargs):
+        """
+        Temporary method to manage the deprecation
         """
-        Sanity checks for X_labels and y_labels.
+        if X is not None or topo_view is not None:
+            warnings.warn(
+                'A dataset is using the old interface that is now deprecated '
+                'and will become officially unsupported as of July 27, 2015. '
+                'The dataset should use the new interface that inherits from '
+                'the dataset class instead of the DenseDesignMatrix class. '
+                'Please refer to pylearn2.datasets.hdf5.py for more details '
+                'on arguments and details of the new '
+                'interface.', DeprecationWarning)
+            return HDF5DatasetDeprecated(filename, X, topo_view, y, load_all,
+                                         cache_size, **kwargs)
+        else:
+            return super(HDF5Dataset, cls).__new__(
+                cls, filename, sources, spaces, aliases, load_all, cache_size,
+                use_h5py, **kwargs)
 
-        Since the np.all test used for these labels does not work with HDF5
-        datasets, we issue a warning that those values are not checked.
+    def __init__(self, filename, sources, spaces, aliases=None, load_all=False,
+                 cache_size=None, use_h5py='auto', **kwargs):
         """
-        if self.X_labels is not None:
-            assert self.X is not None
-            assert self.view_converter is None
-            assert self.X.ndim <= 2
-            if self.load_all:
-                assert np.all(self.X < self.X_labels)
-            else:
-                warnings.warn("HDF5Dataset cannot perform test np.all(X < " +
-                              "X_labels). Use X_labels at your own risk.")
-
-        if self.y_labels is not None:
-            assert self.y is not None
-            assert self.y.ndim <= 2
-            if self.load_all:
-                assert np.all(self.y < self.y_labels)
+        Class constructor
+        """
+        assert isinstance(filename, basestring)
```

Comment:
`basestring` doesn't exist in Python 3. Use `theano.compat.six.string_types` instead.